### PR TITLE
[Java] Switch Java 16 to Java 17 in workflows

### PR DESCRIPTION
**Pull request by luke-hill**
### 🤔 What's changed?

Fix up workflows

### ⚡️ What's your motivation? 

Fix CI

### 🏷️ What kind of change is this?

- :bug: Bug fix (non-breaking change which fixes a defect)

### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
- [ ] I've changed the behaviour of the code
  - [ ] I have added/updated tests to cover my changes.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly.
- [ ] Users should know about my change
  - [ ] I have added an entry to the "Unreleased" section of the [**CHANGELOG**](../blob/main/CHANGELOG.md), linking to this pull request.

----

*This text was originally generated from a [template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/about-issue-and-pull-request-templates), then edited by hand. [You can modify the template here.](https://github.com/cucumber/.github/edit/main/.github/PULL_REQUEST_TEMPLATE.md)*

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 - [Python] Reuse the action cucumber/action-publish-pypi in release ([#147](https://github.com/cucumber/tag-expressions/pull/147))
 - [Ruby] Fixed up remaining simple cops and began to reduce complexity of code ([#158](https://github.com/cucumber/tag-expressions/pull/158))
 
+### Fixed
+- [Java] Altered workflow to use Java 17 LTS over Java 16 ([#167](https://github.com/cucumber/tag-expressions/pull/167))
```
Comment (mpkorstanje) [inline CHANGELOG.md:14]: This doesn't change anything for the end user. So it doesn't have to be in the change log.

Comment (mpkorstanje) [reply CHANGELOG.md:14]: Fixed with e60f74ae0764c52d549cbfe331d8c68fb42e9ebd

Comment (luke-hill) [reply CHANGELOG.md:14]: We do document changes to GHA in our changelog. See the python change by Bjorn recently.

But given Java is your remit I see no reason to debate

Comment (mpkorstanje) [reply CHANGELOG.md:14]: Mmh. The change logs is for notable changes. And I wouldn't consider changes like that notable as it doesn't change how the code works.

**Pull request merged**